crates/re_space_view_spatial/src/ui.rs

```diff
@@ -231,6 +232,72 @@ impl SpatialSpaceViewState {
         Some(())
     }
 
+    fn update_transform3d_lines_heuristics(
+        &self,
+        ctx: &ViewerContext<'_>,
+        ent_path: &EntityPath,
+        entity_properties: &mut re_data_store::EntityPropertyMap,
+    ) {
+        if ctx
+            .store_db
+            .store()
+            .query_latest_component::<Transform3D>(ent_path, &ctx.current_query())
+            .is_none()
+        {
+            return;
+        }
+
+        let mut properties = entity_properties.get(ent_path);
+        if properties.transform_3d_visible.is_auto() {
+            fn path_or_child_has_pinhole(
+                store: &re_arrow_store::DataStore,
+                ent_path: &EntityPath,
+                ctx: &ViewerContext<'_>,
+            ) -> bool {
+                if store
+                    .query_latest_component::<Pinhole>(ent_path, &ctx.current_query())
+                    .is_some()
+                {
+                    return true;
+                } else {
+                    // Any direct child has a pinhole camera?
+                    if let Some(child_tree) = ctx.store_db.entity_db.tree.subtree(ent_path) {
+                        for child in child_tree.children.values() {
+                            if store
+                                .query_latest_component::<Pinhole>(
+                                    &child.path,
+                                    &ctx.current_query(),
+                                )
+                                .is_some()
+                            {
+                                return true;
+                            }
+                        }
+                    }
+                }
+
+                false
+            }
+            // By default show the transform if it is a camera extrinsic or if it's the only component on this entity path.
+            let single_component = ctx
+                .store_db
+                .store()
+                .all_components(&ctx.current_query().timeline, ent_path)
+                .map_or(false, |c| c.len() == 1);
+            properties.transform_3d_visible = EditableAutoValue::Auto(
+                single_component || path_or_child_has_pinhole(ctx.store_db.store(), ent_path, ctx),
+            );
+        }
+
+        if properties.transform_3d_size.is_auto() {
+            properties.transform_3d_size =
+                // Reuse `auto_size_world_heuristic` with a scale factor so we don't reinvent the wheel on scene based size heuristic.
+                EditableAutoValue::Auto(self.auto_size_world_heuristic() * 5.0);
```

Comment:
I'm all for reusing logic, but `auto_size_world_heuristic` (which lacks a docstring btw) gets smaller as the number of entities grows. This makes sense for something like a point cloud (few points -> big points, many points -> small points), but it doesn't really make sense for axis arrows, does it?